src/cephadm/cephadm

```diff
@@ -1078,6 +1078,38 @@ def infer_fsid(func):
 
     return _infer_fsid
 
+def infer_config(func):
+    """
+    If we find a MON daemon, use the config from that container
+    """
+    @wraps(func)
+    def _infer_config():
+        if args.config:
+            logger.debug('Using specified config: %s' % args.config)
+            return func()
+        config = None
+        if args.fsid:
+            name = args.name
+            if not name:
+                daemon_list = list_daemons(detail=False)
+                for daemon in daemon_list:
+                    if daemon['name'].startswith('mon.'):
+                        name = daemon['name']
+                        break
+            if name:
+                config = '/var/lib/ceph/{}/{}/config'.format(args.fsid, name)
+        if config:
+            logger.info('Inferring config %s' % config)
+            args.config = config
+        elif os.path.exists(SHELL_DEFAULT_CONF):
+            logger.debug('Using default config: %s' % SHELL_DEFAULT_CONF)
+            args.config = SHELL_DEFAULT_CONF
+        else:
+            raise Error('Cannot infer a config, one must be specified')
```

Comment:
```
2020-04-26T14:23:46.525 INFO:tasks.workunit.client.0.smithi183.stderr:+ grep 'ceph version'
2020-04-26T14:24:11.498 INFO:tasks.workunit.client.0.smithi183.stdout:ceph version 16.0.0-1082-g28c7ba68ae (28c7ba68ae17612394a47291f42995d4c893a440) pacific (dev)
2020-04-26T14:24:11.498 INFO:tasks.workunit.client.0.smithi183.stderr:+ systemctl status docker
2020-04-26T14:24:11.507 INFO:tasks.workunit.client.0.smithi183.stderr:Unit docker.service could not be found.
2020-04-26T14:24:11.508 INFO:tasks.workunit.client.0.smithi183.stderr:+ sudo /tmp/tmp.aA4DdzJJSf/cephadm --image docker.io/ceph/daemon-base:latest-master-devel shell --fsid 00000000-0000-0000-0000-0000dea
dbeef -- ceph -v
2020-04-26T14:24:11.508 INFO:tasks.workunit.client.0.smithi183.stderr:+ grep 'ceph version'
2020-04-26T14:24:11.630 INFO:tasks.workunit.client.0.smithi183.stderr:ERROR: Cannot infer a config, one must be specified
2020-04-26T14:24:11.640 INFO:tasks.workunit.client.0.smithi183.stderr:+ cleanup
```

http://pulpito.ceph.com/kchai-2020-04-26_13:55:50-rados-wip-kefu-testing-2020-04-26-1314-distro-basic-smithi/4986840/

Reply:
@tchaikov I've updated this PR to **NOT** throw an error when config **cannot** be inferred. Please test again, thanks.